x-pack/plugins/cloud_security_posture/public/components/fleet_extensions/gcp_credentials_form/gcp_credential_form.tsx

```diff
@@ -581,15 +589,41 @@ export const GcpInputVarFields = ({
           </EuiFormRow>
         )}
         {credentialsTypeValue === credentialJSONValue && credentialJSONFields && (
-          <EuiFormRow fullWidth label={gcpField.fields['gcp.credentials.json'].label}>
-            <EuiTextArea
-              data-test-subj={CIS_GCP_INPUT_FIELDS_TEST_SUBJECTS.CREDENTIALS_JSON}
-              id={credentialJSONFields.id}
-              fullWidth
-              value={credentialJSONFields.value || ''}
-              onChange={(event) => onChange(credentialJSONFields.id, event.target.value)}
-            />
-          </EuiFormRow>
+          <div
+            css={css`
+              width: 100%;
+              .euiFormControlLayout,
+              .euiFormControlLayout__childrenWrapper,
+              .euiFormRow,
+              input {
+                max-width: 100%;
+                width: 100%;
+              }
+            `}
+          >
+            <EuiSpacer size="m" />
+            <EuiFormRow fullWidth label={gcpField.fields['gcp.credentials.json'].label}>
+              <Suspense fallback={<EuiLoadingSpinner size="l" />}>
+                <LazyPackagePolicyInputVarField
+                  data-test-subj={CIS_GCP_INPUT_FIELDS_TEST_SUBJECTS.CREDENTIALS_JSON}
+                  varDef={{
+                    ...findVariableDef(packageInfo, credentialJSONFields.id)!,
+                    required: true,
+                    type: 'textarea',
+                    secret: true,
+                    full_width: true,
+                  }}
+                  value={credentialJSONFields.value || ''}
+                  onChange={(value) => {
+                    onChange(credentialJSONFields.id, value);
+                  }}
+                  errors={[]}
```

Comment:
im not familiar with this component, is this on purpose? what does this effects?

Reply:
I believe this just show a error string message, originally we use [] as placeholder because other team that uses this component seems to use var that our component dont have